iib/workers/tasks/git_utils.py

```diff
@@ -0,0 +1,255 @@
+            )
+            log.info(git_status)
+            commit_and_push(
+                request_id,
+                local_repo_dir,
+                repo_url,
+                branch,
+                f"IIB: Update for request id {request_id} (overwrite_from_index)",
+            )
+        finally:
+            # tempfile should have done this already, but just in case
+            if os.path.exists(local_repo_dir):
+                shutil.rmtree(local_repo_dir)
+                log.debug("Cleaned up local Git repository %s", local_repo_dir)
+
+
+def commit_and_push(
+    request_id: int,
+    local_repo_path: str,
+    repo_url: str,
+    branch: str,
+    commit_message: Optional[str] = None,
+) -> None:
+    """
+    Commit and push locally staged changes to remote Git repo.
+
+    :param int request_id: The ID of the IIB request.
+    :param str local_repo_path: Path to local git repository where changes have been staged.
+    :param str commit_message: Custom commit message. If None, a default message is used.
+    :raises IIBError: If a Git operation fails.
+    """
+    # Commit updates
+    final_commit_message = commit_message or (
+        f"IIB: Update for request id {request_id} (overwrite_from_index)"
+    )
+    commit_output = run_cmd(
+        ["git", "-C", local_repo_path, "commit", "-m", final_commit_message],
+        exc_msg="Error committing changes",
+    )
+    log.info(commit_output)
+
+    # Push updates
+    log.info("Pushing changes to %s branch of %s", branch, repo_url)
+    push_output = run_cmd(
+        ["git", "-C", local_repo_path, "push", "--force", "origin", branch],
+        exc_msg=f"Error pushing changes to git repo {repo_url}",
+    )
+    log.info(push_output)
+
+
+def get_gitlab_url(from_index) -> str:
+    """
+    Get Git repository URL from iib_web_index_to_gitlab_push_map.
+
+    :param str from_index: from_index image pull spec.
+    :return: Git URL.
+    :rtype: str
+    :raises IIBError: If no mapping found.
+    """
+    index_image = ImageName.parse(from_index)
+    index_no_tag = f"{index_image.registry}/{index_image.namespace}/{index_image.repo}"
+    index_repo_map = get_worker_config()['iib_web_index_to_gitlab_push_map']
+    gitlab_url = index_repo_map.get(index_no_tag, None)
+    if not gitlab_url:
+        raise IIBError(f"No 'iib_web_index_to_gitlab_push_map' found for image {from_index}")
+    return gitlab_url
+
+
+def get_gitlab_token(git_repo) -> Tuple[str, str]:
+    """
+    Get Git repository token from iib_index_configs_gitlab_tokens_map.
+
+    :param str git_repo: Git repository URL.
+    :return: token name, token value.
+    :rtype: Tuple[str, str]
+    :raises IIBError: If no token found for Git repository.
+    """
+    git_token_map = get_worker_config()['iib_index_configs_gitlab_tokens_map']
+    if git_repo not in git_token_map:
+        raise IIBError(f"No 'iib_index_configs_gitlab_tokens_map' found for Git repo {git_repo}")
```

Comment:
Personally, I would rephrase this something like this: "Missing key {git repo} in [variable_name]">

But this is just a preference; I will leave this up to you.

Reply:
Fixed per your suggestion